eppo/src/androidTest/java/cloud/eppo/android/EppoClientTest.java

```diff
@@ -249,26 +244,22 @@ private void runTestCases() {
             }
             System.out.println("We ran this many tests: " + testsRan);
             assertTrue("Did not run any test cases", testsRan > 0);
-        } catch (Exception e) {
+        } catch (IOException e) {
             throw new RuntimeException(e);
         }
     }
 
     @Test
     public void testCachedAssignments() {
-        try {
-            // First initialize successfully
-            initClient(TEST_HOST, false, true, false); // ensure cache is populated
+        // First initialize successfully
+        initClient(TEST_HOST, false, true, false); // ensure cache is populated
 
-            // wait for a bit since cache file is loaded asynchronously
-            System.out.println("Sleeping for a bit to wait for cache population to complete");
-            Thread.sleep(10000);
+        // wait for a bit since cache file is loaded asynchronously
+        waitForNonNullAssignment();
```

Comment:
This test seemed flakey for me so I changed the hard-coded wait to use a checking one another test uses too